Tests/SonarQube.TeamBuild.PostProcessor.Tests/Infrastructure/MockCodeCoverageProcessor.cs

```diff
@@ -7,42 +7,64 @@
 using SonarQube.TeamBuild.Integration;
 using SonarQube.Common;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
 
 namespace SonarQube.TeamBuild.PostProcessor.Tests
 {
     internal class MockCodeCoverageProcessor : ICoverageReportProcessor
     {
-        private bool methodCalled;
+        private bool processCoverageMethodCalled;
+        private bool initalizedCalled;
 
         #region Test helpers
 
-        public bool ValueToReturn { get; set; }
+        public bool ProcessValueToReturn { get; set; }
+        public bool InitialiseValueToReturn { get; set; }
 
         #endregion
 
         #region ICoverageReportProcessor interface
 
-        public bool ProcessCoverageReports(AnalysisConfig context, TeamBuildSettings settings, ILogger logger)
+        public bool Initialise(AnalysisConfig context, TeamBuildSettings settings, ILogger logger)
         {
-            Assert.IsFalse(this.methodCalled, "Expecting ProcessCoverageReports to be called only once");
-            this.methodCalled = true;
-            return ValueToReturn;
+            Assert.IsFalse(this.initalizedCalled, "Expecting Initialise to be called only once");
+            this.initalizedCalled = true;
+            return InitialiseValueToReturn;
+        }
+
+        public bool ProcessCoverageReports()
+        {
+            Assert.IsFalse(this.processCoverageMethodCalled, "Expecting ProcessCoverageReports to be called only once");
+            Assert.IsTrue(this.initalizedCalled, "Expecting Initialise to be called first");
+            this.processCoverageMethodCalled = true;
+            return ProcessValueToReturn;
         }
 
         #endregion
 
         #region Checks
 
-        public void AssertExecuted()
+        public void AssertExecutedCalled()
+        {
+            Assert.IsTrue(this.processCoverageMethodCalled, "Expecting the sonar-runner to have been called");
+        }
+
+        public void AssertExecutedNotCalled()
```

Comment:
FYI Code coverage requires VS2013 Ultimate or Premium / VS2015 Enterprise.
It isn't available in VS2013 Professional.
See https://msdn.microsoft.com/en-us/library/dd537628(v=vs.120).aspx